core/src/test/java/org/elasticsearch/search/aggregations/bucket/range/InternalBinaryRangeTests.java

```diff
@@ -40,26 +42,28 @@
     public void setUp() throws Exception {
         super.setUp();
 
-        final int numRanges = randomIntBetween(1, 10);
-        ranges = new ArrayList<>(numRanges);
+        final int numRanges = randomNumberOfBuckets();
+        List<Tuple<BytesRef, BytesRef>> listOfRanges = new ArrayList<>(numRanges);
 
         for (int i = 0; i < numRanges; i++) {
             BytesRef[] values = new BytesRef[2];
             values[0] = new BytesRef(randomAlphaOfLength(15));
             values[1] = new BytesRef(randomAlphaOfLength(15));
             Arrays.sort(values);
-            ranges.add(Tuple.tuple(values[0], values[1]));
+            listOfRanges.add(Tuple.tuple(values[0], values[1]));
         }
 
         if (randomBoolean()) {
-            ranges.add(Tuple.tuple(null, new BytesRef(randomAlphaOfLength(15))));
+            listOfRanges.add(Tuple.tuple(null, new BytesRef(randomAlphaOfLength(15))));
         }
         if (randomBoolean()) {
-            ranges.add(Tuple.tuple(new BytesRef(randomAlphaOfLength(15)), null));
+            listOfRanges.add(Tuple.tuple(new BytesRef(randomAlphaOfLength(15)), null));
         }
         if (randomBoolean()) {
-            ranges.add(Tuple.tuple(null, null));
+            listOfRanges.add(Tuple.tuple(null, null));
         }
+        Collections.shuffle(listOfRanges, random());
+        ranges = Collections.unmodifiableList(listOfRanges.stream().limit(maxNumberOfBuckets()).collect(Collectors.toList()));
```

Comment:
Nit: If we do the shuffling, can we fill the listOfRanges with the random edge-cases (the null cases) first and then fill up the rest up to the available amount of buckets with other random values? That way we are sure to include the edge cases in 50% of the cases.
Its not super important here but I think it would be easier to understand than the truncation of the list.